rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbMsgCountNodeTest.java

```diff
@@ -0,0 +1,199 @@
+import org.thingsboard.server.common.msg.TbMsg;
+import org.thingsboard.server.common.msg.TbMsgMetaData;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+@ExtendWith(MockitoExtension.class)
+public class TbMsgCountNodeTest {
+
+    private final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+    private TbMsgCountNode node;
+    private TbMsgCountNodeConfiguration config;
+    private CountDownLatch awaitTellSelfLatch;
+
+    @Mock
+    private TbContext ctxMock;
+
+    @BeforeEach
+    public void setUp() {
+        node = new TbMsgCountNode();
+        config = new TbMsgCountNodeConfiguration();
+    }
+
+    @Test
+    public void givenIncomingMsgs_whenOnMsg_thenSendsMsgWithMsgCount() throws TbNodeException, InterruptedException {
+        config.setInterval(10);
+        config.setTelemetryPrefix("count");
+        var configuration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+
+        int numberOfMsgs = 7;
+        int wantedNumberOfTellSelfInvocation = 2;
+
+        awaitTellSelfLatch = new CountDownLatch(wantedNumberOfTellSelfInvocation);
+        invokeTellSelf(wantedNumberOfTellSelfInvocation, config.getInterval());
+
+        TbMsg tickMsg = TbMsg.newMsg(null, TbMsgType.MSG_COUNT_SELF_MSG, null, null, TbMsgMetaData.EMPTY, TbMsg.EMPTY_STRING);
+        when(ctxMock.newMsg(any(), any(TbMsgType.class), any(), any(), any(), any())).thenReturn(tickMsg);
+
+        node.init(ctxMock, configuration);
+
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, null, TbMsgMetaData.EMPTY, TbMsg.EMPTY_STRING);
+        for (int i = 0; i < numberOfMsgs; i++) {
+            node.onMsg(ctxMock, msg);
+            Thread.sleep(2000);
+        }
+
+        awaitTellSelfLatch.await();
+
+        verify(ctxMock, times(numberOfMsgs)).ack(any());
+        ArgumentCaptor<TbMsg> newMsgCaptor = ArgumentCaptor.forClass(TbMsg.class);
+        verify(ctxMock, times(wantedNumberOfTellSelfInvocation)).enqueueForTellNext(newMsgCaptor.capture(), eq(TbNodeConnectionType.SUCCESS));
+
+        List<TbMsg> resultMsgs = newMsgCaptor.getAllValues();
+        Assertions.assertEquals(wantedNumberOfTellSelfInvocation, resultMsgs.size());
+
+        TbMsg firstMsg = resultMsgs.get(0);
+        assertThat(firstMsg.getData()).isEqualTo("{\"count_null\":5}");
```

Comment:
please mock ctx.getServiceId() to something like tb-rule-engine. then the data will contain count_tb-rule-engine